src/IO/H5/VolumeData.cpp

```diff
@@ -299,14 +397,22 @@ std::vector<std::string> VolumeData::list_tensor_components(
     // NOLINTNEXTLINE(bugprone-unused-return-value)
     alg::remove(tensor_components, data_name);
   };
+
+  const auto search_result = alg::find(tensor_components, "pole_connectivity");
+  const auto last = tensor_components.end();
```

Comment:
you never use `last` again, so just inline it? You also never use `search_result` again. Can you move this after the `remove_data_names`? I would've found it easier to understand why you're setting this variable

Reply:
The test checking `read_components` fails when I make this change. I think the problem is that search_result can't find pole_connectivity after the data name is removed, so the wrong number of components are erased